doc/guides/west/without-west.rst

```diff
@@ -3,30 +3,79 @@
-This section applies to Zephyr users who fall into one or more of these
-categories:
+This page provides information on using Zephyr without west. This is
+not recommended for beginners due to the extra effort involved. In
+particular, you will have to do work "by hand" to replace these
+features:
 
-- Already use a multi-repository management system (for example Google repo)
+- cloning the additional source code repositories used by Zephyr in
+  addition to the main zephyr repository, and keeping them up to date
+- specifying the locations of these repositories to the Zephyr build
+  system
+- flashing and debugging without understanding detailed usage of the
+  relevant host tools
 
-- Do not need additional functionality beyond compiling and linking
+.. note::
 
-In order to obtain the Zephyr source code and build it without the help of
-west you will need to manually clone the repositories listed in the
-`manifest file`_ one by one, at the path indicated in it.
+   If you have previously installed west and want to stop using it,
+   uninstall it first:
+
+   .. code-block:: console
+
+      pip3 uninstall west
+
+   Otherwise, Zephyr's build system will find it and may try to use
+   it.
+
+Getting the Source
+------------------
+
+In addition to downloading the zephyr source code repository itself,
+you will need to manually clone the additional projects listed in the
+:term:`west manifest` file inside that repository.
 
 .. code-block:: console
 
    mkdir zephyrproject
    cd zephyrproject
    git clone https://github.com/zephyrproject-rtos/zephyr
-   # clone additional repositories listed in the manifest file,
-   # and check out the specified revisions
+   # clone additional repositories listed in zephyr/west.yml,
+   # and check out the specified revisions as well.
+
+As you pull changes in the zephyr repository, you will also need to
+maintain those additional repositories, adding new ones as necessary
+and keeping existing ones up to date at the latest revisions.
+
+Specifying Modules
+------------------
 
-If you want to manage Zephyr's repositories without west but still need to
-use west's additional functionality (flashing, debugging, etc.) it is possible
-to do so by manually creating an installation:
+If you have west installed, the Zephyr build system will use it to set
+:ref:`ZEPHYR_MODULES <important-build-vars>`. If you don't have west
+installed and your application does not need any of these
+repositories, the build will still work.
+
+If you don't have west installed and your application *does* need one
+of these repositories, you must set :makevar:`ZEPHYR_MODULES`
+yourself. You can do this at CMake time, e.g. with a command line like
+``cmake -DZEPHYR_MODULES=path/to/repository1;path/to/repository2
+[...]``. You will need to maintain changes to this list as additional
+code is migrated out of zephyr's :file:`ext` directory into separate
+Git repositories.
```

Comment:
Question: Regarding the zephyr.modules option, would it be worth adding 2 lines here?

Reply:
> Question: Regarding the zephyr.modules option, would it be worth adding 2 lines here?

I think you mean manually creating a zephyr_modules.txt file in the build directory before running CMake?

I actually don't want to document that right now, because I looked at the code and noticed that Kconfig.modules will always be overwritten with `# No west and no modules` if ZEPHYR_MODULES is not set. This means that users can set up additional modules for CMake purposes, but those modules won't be able to have their own Kconfig options.

Let's fix that problem after the 1.14 release so that if Kconfig.modules is already in the build directory, the build system will just use it. This will let users create build/zephyr_modules.txt and build/Kconfig.modules before running CMake with feature parity compared to setting ZEPHYR_MODULES by hand.

Reply:
More ideas in PR #15042

Reply:
> More ideas in PR #15042

Yep, but that would have to come after LTS; since this is a docs-only patch, it can go in before the release.